File: solutions/haskell/lucians-luscious-lasagna/1/src/LuciansLusciousLasagna.hs
module LuciansLusciousLasagna (elapsedTimeInMinutes, expectedMinutesInOven, preparationTimeInMinutes) where

expectedMinutesInOven :: Integer
expectedMinutesInOven = 40

preparationTimeInMinutes :: Num a => a -> a
preparationTimeInMinutes layers = layers * 2

elapsedTimeInMinutes :: Num a => a -> a -> a
elapsedTimeInMinutes layers minutes_in_oven = preparationTimeInMinutes layers + minutes_in_oven

File: solutions/fsharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40
let remainingMinutesInOven = (-) expectedMinutesInOven
let preparationTimeInMinutes = (*) 2

let elapsedTimeInMinutes layers minutes =
    minutes + preparationTimeInMinutes layers
